### ISSUE TITLE
Support profile updates

### ISSUE THREAD

**Issue by gRegorLove** [snarfed/bridgy-fed#376]
It would be nice if BF could send profile update activities when h-card bio, photo, and link(s) change. I recently changed the h-card on my homepage and instances aware of me prior to that still have my old photo. You [mentioned in chat](https://chat.indieweb.org/dev/2023-01-03#t1672777500729600) that BF would need to poll for updates. I also wonder if an on-demand option could work via a webmention trigger? I have not thought through this in-depth, but maybe microformats like:

<code>&lt;a href="https://gregorlove.com" class="u-edit-of"&gt;updated profile&lt;/a&gt;</code>

(Originally published at: https://gregorlove.com/2023/01/support-profile-updates/)

**Comment by snarfed** [comment]
Yes! Great idea. Webmention trigger could work, or I could just add a 🔄 button to user pages.

**Comment by snarfed** [comment]
Oh, but the catch is, we'd need to know all the instances to send the profile `Update` to. I guess I can send it to all of your followers' instances? That's a subset of all instances aware of you, I can try to collect that list, but it will still be a subset. 🤔

**Comment by gRegorLove** [comment]
I was thinking only instances that someone had followed me from would have the local cache of my profile. Is it also on instances I have followed someone on? If so, I think followers-only seems like a good start.

(Originally published at: https://gregorlove.com/2023/01/i-was-thinking-only/)

**Comment by snarfed** [comment]
Done! Docs: https://fed.brid.gy/docs#update-profile

**Comment by SievaFliesAlone** [comment]
@snarfed I'm not sure if I should open a new issue, but the profile update never worked for me —— my profile description is still the same as it was when I first bridged it a couple of years ago

**Comment by snarfed** [comment]
@SievaFliesAlone hmm, sorry about that! Are you on the fediverse? If so, when @bsky.brid.gy tried to follow you, did you accept that follow?

You can also trigger profile updates manually, https://fed.brid.gy/docs#update-profile , but you're right that ideally we should do it automatically.

**Comment by SievaFliesAlone** [comment]
I'm on the fedi. @bsky.brid.gy and I follow each other. The posts get across, but profile description update doesn't work, including the manual option.

**Comment by snarfed** [comment]
@SievaFliesAlone oof, that does sound broken. Sorry! What's your fediverse account?

**Comment by SievaFliesAlone** [comment]
[redacted]

**Comment by snarfed** [comment]
Thanks! I've nudged your account, its profile updates should work now.

**Comment by golemwire** [comment]
Hello @snarfed ! I'm having the same issue, as described by SievaFliesAlone. I have a Fediverse account, at `@golemwire@social.golemwire.com`, and its [Bluesky page](https://bsky.app/profile/golemwire.social.golemwire.com.ap.brid.gy) doesn't have my PFP, bio, or even username at all... I've clicked the manual brid.gy refresh button at https://fed.brid.gy/ap/@golemwire@social.golemwire.com as well, but to no avail.

Notes:

*	Social.golemwire.com runs Snac2.
*	The posts from social.golemwire.com appear to be bridging OK.
*	My previous fedi account, `@golemwire@fosstodon.org`, was bridged to Bluesky, successfully ([Bluesky page](https://bsky.app/profile/golemwire.fosstodon.org.ap.brid.gy)). I migrated from it to my current account (the one with the profile bridging issues), but the migration didn't seem to work (I don't have my previous Bluesky followers).
	*	Note! I had migrated to `@golemwire@social.golemwire.com` very shortly after creating it. Perhaps brid.gy wasn't yet aware of its existence when I initiated the move, triggering all these issues?

**Comment by golemwire** [comment]
Another note: I just changed my Bluesky handle to `@golemwire.com` from `@golemwire.social.golemwire.com.ap.brid.gy`.\
I clicked the refresh button again, but still no good (bio, pfp, and username are still missing (though the handle did update)).

**Comment by golemwire** [comment]
The problem went away when I did this: I edited my account on social.golemwire.com, then clicked refresh on https://fed.brid.gy/ap/@golemwire@social.golemwire.com within a minute, then it updated. Not sure what happened, but I'm glad it is working now :)

@snarfed TLDR: It appears that, possibly, migrating Fedi accounts can cause some issues with the bio not updating, which can be fixed by renaming the Bluesky account, updating your Fedi bio, then clicking refresh on fed.brid.gy.

The migration feature still hasn't worked, but I'll have to open an issue for that (or find an existing one).

Bizarrely, Bluesky now says my account is 11 minutes old (??) even though a post from 6d ago is up.

**Comment by snarfed** [comment]
Hey @golemwire, sorry for the trouble!

The main reason your bridged Bluesky profile didn't get populated was Snac2's pinned post handling. It puts `Create` activities into your actor's `featured` collection, which is unusual; usually that contains objects like `Note`s directly. https://browser.pub/https%3A%2F%2Fsocial.golemwire.com%2Fgolemwire%2Ffeatured . (They're also not fetchable by id, but that's not as big a deal.)

`featured` isn't part of AS2 Core or Vocab, and [it's only lightly specified by Mastodon themselves](https://docs.joinmastodon.org/spec/activitypub/#featured), so I can't conclusively say that Snac2 is wrong, but you may see similar interop issues in other places beyond Bridgy Fed.

**Comment by snarfed** [comment]
As for your migration, looks like that happened over 30d ago? I don't have logs that far back, but our current fediverse migration handling expects that the new account isn't already bridged itself. Looks like you may have enabled the bridge on your new account before you migrated your old account?

**Comment by golemwire** [comment]
@snarfed Thank you for all your help!

I had enabled the bridge via following the follow-back account. So, I figure while I was in the process of moving accounts, my follows were being transferred over and during that process, my new account followed the follow-back account, which triggered the follow-back and thus the issue.

I followed the guide in [snac(8)](https://manpages.ubuntu.com/manpages/questing/man8/snac.8.html#Migrating_from_Mastodon_to_snac) to perform the migration. It looks like it had me import my followers list before performing the actual Move....\
If this is an error in the Snac2 documentation, I can open an issue on its repository.

(I had few followers on the old account anyway, and can DM them about my new account anyway; I'm not asking you to manually fix it if it takes any effort for you. I'm mainly hoping to help you find a potential bug.)
